src/nft/components/bag/BagFooter.tsx

```diff
@@ -377,105 +379,78 @@ export const BagFooter = ({ setModalIsOpen, eventProperties }: BagFooterProps) =
     helperTextColor,
     handleClick,
     buttonColor,
-  } = useMemo(() => {
-    let handleClick: (() => void) | (() => Promise<void>) = fetchAssets
-    let buttonText = <Trans>Something went wrong</Trans>
-    let disabled = true
-    let warningText = undefined
-    let warningTextColor = theme.accentWarning
-    let helperText = undefined
-    let helperTextColor = theme.textSecondary
-    let buttonColor = theme.accentAction
-    let buttonTextColor = theme.accentTextLightPrimary
-
+  } = useMemo((): BuyButtonState => {
     if (connected && chainId !== SupportedChainId.MAINNET) {
-      handleClick = () => switchChain(connector, SupportedChainId.MAINNET)
-      buttonText = <Trans>Switch networks</Trans>
```

Comment:
The new refactor doesn't check for `BuyButtonStates.NOT_SUPPORTED_CHAIN` ie I'm on Polygon and getting the 'Connect Wallet' text
![Screen Shot 2023-05-23 at 09 45 19 ](https://github.com/Uniswap/interface/assets/11512321/4baf09a3-0f76-430a-bb54-044b48f16f02)